app_spec/zomes/simple/code/src/lib.rs

```diff
@@ -63,6 +63,10 @@ pub fn handle_get_my_links(base: Address) -> ZomeApiResult<GetLinksResult> {
     hdk::get_links(&base, LinkMatch::Exactly("authored_posts"), LinkMatch::Any)
 }
 
+pub fn handle_test_emit_signal(message: String) -> ZomeApiResult<()> {
+    hdk::emit_signal("test-signal", JsonString::from_json(&format!("{{\"message\": {}}}", message)))
```

Comment:
I think a good improvement would be to use a struct that implements DefaultJson rather than this string formatting. 

e.g. 
```rust
#[derive(..., DefaultJson)]
struct SignalPayload {
   message: String
}
...
let payload = SignalPayload{message}
hdk::emit_signal("test-signal", payload.into())
```

Only because people are going to copy what is in the app spec and I don't think this is the best practice.

Reply:
Yes, good point, will add that!